### PR TITLE
[485305]: Implement TODO (step 2)

### PR DESCRIPTION
**Pull request by zacikpa**

### CODE REVIEW

**Comment by zacikpa** [comment]
Done.

**Comment by ph4r05** [comment]
One additional thing: pls add commit info file from the assignment here, as a comment (plaintext, no attachments)

**Comment by zacikpa** [comment]
STEP 1:

I thought that the issue with Python unittests discovery was something we had to do by ourselves,
so I did it here already because we weren't supposed to read step 2.

- These are on pr/step1-old:

622fa225f004a4c0130b37395f2dcb7fde727cd9 - Implement all TODO functions
d69518264482144c21a7702db2f23d2340a39412 - Import sqrt from math package
76e53246bc1f20586ef155c2ac0dada6a7a060a5 - Deprecate magic __cmp__ method 
b7fa14861118d6e9392959c76d4ef10c3a2c0ae4 - Add tests for added functionality
65aec8ce565c79c55d4868fd8eff6ccf3105e042 - Fix python unittests discovery
fa81dbe8e217faadda178b2f0933ca80ed631784 - Update ubuntu version in Travis
0df3c3518788bdac0b155b77b5279a4b69ec38d7 - Add simple docs to README.md
8db76e45cea56065dee22fc83138139413d35722 - Add Travis badge to README.md
e9e9e867fef7b0b01b813833e9bcf241f5525110 - Add a simple comment to the __sub__ method

- On pr/step1:

5b94d6b0ef431c24bc36c75bd5808cc7420c51b1 - Implement all TODO functions (squashed)

- When squashing, I forgot to sign the new commit (It's rather counter-intuitive), so I made a new one.
- Yes, it is a force push. :( 
- It only adds the signature, however.

536e67a8872ded157760aad40dac7336dbdaff70 - Implement all TODO functions (signed)


STEP 2:

- These are on pr/step2-old:

0e8230d499c9f4d293ad26610eb3ef920cffc15b - Implement all TODO functions (rebased)
280963241c0228095dbc1b1bb88f2bac3e18087a - Implement remaining TODOs (step 2)
8432ea040654630c8e957e0c9de9025796dd2573 - Add tests for newly added functions
d1009fc3be88420a588c933a70c8caf0db032ead - Add missing description to README.md

- On pr/step2:

f9ddacefa80bb6c220b8469cc6924f1b81c110fa - Add matrix and implement remaining TODO functions (squashed)

- Again, I realized too late I have to sign when rebasing.
- This is the final commit in the PR 2.

fbbda9cc3cdaf9cc7202def17de53bc42506884b - Add matrix and implement remaining TODO functions (signed)

**Review by ph4r05 — APPROVED** [review]

File: ossdev/__init__.py [hunk 1]
```diff
@@ -21,51 +24,178 @@ def set(self, arr):
...
-        # TODO: implement
-        return None
+        self.d[key] = value
+
+    # magic __cmp__ function is deprecated since Python3
+    # def __cmp__(self, other):
+    #     return (self.length() > other.length()) - (self.length() < other.length())
+
+    def __eq__(self, other):
+        return self.length() == other.length()
```
Comment (ph4r05) [inline ossdev/__init__.py:57]: comparison just by length is not enough!

Comment (ph4r05) [reply ossdev/__init__.py:57]: You dont have to fix it. Just a note.

Comment (ph4r05) [reply ossdev/__init__.py:57]: If you choose to remove `__cmp__` you should properly implement other comparators. 
E.g., with https://docs.python.org/3/library/functools.html#functools.total_ordering

**Comment by ph4r05** [comment]
Approved, HW done, closing

**Pull request closed without merging**